projects/packages/waf/src/class-waf-rules-manager.php

```diff
@@ -74,59 +76,76 @@ public static function schedule_rules_cron() {
 	public static function update_rules_cron() {
 		Waf_Constants::define_mode();
 		if ( ! Waf_Runner::is_allowed_mode( JETPACK_WAF_MODE ) ) {
-			return new WP_Error( 'waf_cron_update_failed', 'Invalid firewall mode.' );
+			return new WP_Error( 'waf_invalid_mode', 'Invalid firewall mode.' );
 		}
 
 		try {
 			self::generate_automatic_rules();
 			self::generate_ip_rules();
 			self::generate_rules();
-		} catch ( \Exception $e ) {
-			return new WP_Error( 'waf_cron_update_failed', $e->getMessage() );
+		} catch ( Waf_Exception $e ) {
+			return $e->get_wp_error();
 		}
 
 		update_option( self::RULE_LAST_UPDATED_OPTION_NAME, time() );
 		return true;
 	}
 
+	/**
+	 * Re-activate the WAF any time an option is added or updated.
+	 *
+	 * @return bool|WP_Error True if re-activation is successful, WP_Error on failure.
+	 */
+	public static function reactivate_on_rules_option_change() {
+		try {
+			Waf_Runner::activate();
+		} catch ( Waf_Exception $e ) {
+			return $e->get_wp_error();
+		}
+
+		return true;
+	}
+
 	/**
 	 * Updates the rule set if rules version has changed
 	 *
-	 * @return bool|WP_Error True if rules update is successful, WP_Error on failure.
+	 * @throws Waf_Exception If the firewall mode is invalid.
+	 * @throws Waf_Exception If the rules update fails.
+	 *
+	 * @return void
 	 */
 	public static function update_rules_if_changed() {
 		Waf_Constants::define_mode();
 		if ( ! Waf_Runner::is_allowed_mode( JETPACK_WAF_MODE ) ) {
-			return new WP_Error( 'waf_update_failed', 'Invalid firewall mode.' );
+			throw new Waf_Exception( 'Invalid firewall mode.' );
 		}
 		$version = get_option( self::VERSION_OPTION_NAME );
 		if ( self::RULES_VERSION !== $version ) {
-			update_option( self::VERSION_OPTION_NAME, self::RULES_VERSION );
-
 			try {
 				self::generate_automatic_rules();
 				self::generate_ip_rules();
 				self::generate_rules();
-			} catch ( \Exception $e ) {
-				return new WP_Error( 'waf_update_failed', $e->getMessage() );
+			} catch ( Waf_Exception $e ) {
+				throw $e;
```

Comment:
We could also just not have the try-catch block here I think? (I cannot add a suggestion here cause you did not change all the lines)

Reply:
Resolved via https://github.com/Automattic/jetpack/commit/c1d3effd93c8fd23355781bca6b324383c18d498

Reply:
This particular instance of the catch/re-throw is still there.

Reply:
Whoops... [370a517](https://github.com/Automattic/jetpack/pull/29108/commits/370a5177214804fa6a9e88c91d19815fe4aece6a)